src/vs/workbench/contrib/terminal/electron-sandbox/terminal.contribution.ts

```diff
@@ -25,3 +31,40 @@ const workbenchRegistry = Registry.as<IWorkbenchContributionsRegistry>(Workbench
 // terminal while connecting to the remote.
 registerWorkbenchContribution2(LocalTerminalBackendContribution.ID, LocalTerminalBackendContribution, WorkbenchPhase.BlockStartup);
 workbenchRegistry.registerWorkbenchContribution(TerminalNativeContribution, LifecyclePhase.Restored);
+
+registerContextualInstanceAction({
+	id: TerminalCommandId.RevealInFinder,
+	title: localize2('workbench.action.terminal.revealInFinder', 'Reveal in Finder'),
+	run: async (instance, c, accessor) => {
+		const nativeHostService = accessor.get(INativeHostService);
+		// TODO: cwd is empty when the terminal is restore from editor.
+		// const cwd = instance.cwd || instance.initialCwd;
```

Comment:
For restored editors I think this is tracked in another issue. For using the initialCwd, I'm not so sure we should be doing this as the command would then not work as expected. Instead of providing a potentially inconsistent feature we could just only enable this when it will work (ie. `CwdDetectionCapability` exists for the terminal)